lib/spack/spack/spec.py

```diff
@@ -1408,6 +1410,20 @@ def dag_hash_bit_prefix(self, bits):
         """Get the first <bits> bits of the DAG hash as an integer type."""
         return base32_prefix_bits(self.dag_hash(), bits)
 
+    def full_hash(self, length=None):
```

Comment:
Still needs test coverage.  The tests seem to call `package_hash`, but this doesn't.  Why?

Reply:
The tests were for `package_hash` directly. This calls `content_hash` which ends up calling `package_hash` later. Will add tests to this commit.